crates/optimism/chainspec/src/lib.rs

```diff
@@ -194,6 +199,66 @@ impl OpChainSpec {
     pub fn from_genesis(genesis: Genesis) -> Self {
         genesis.into()
     }
+
+    fn get_or_init_sealed_genesis_header(&self) -> &SealedHeader {
+        self.inner.genesis_header.get_or_init(|| {
+            let header = self.make_genesis_header();
+            if self.is_optimism_mainnet() {
+                // for OP mainnet we have to do this because the genesis header can't be
+                // properly computed from the genesis.json file due to OVM history
+                return SealedHeader::new(header, OP_MAINNET_GENESIS_HASH)
+            }
+            SealedHeader::seal_slow(header)
+        })
+    }
+
+    /// Assembles genesis header from genesis file.
+    fn make_genesis_header(&self) -> Header {
+        // If London is activated at genesis, we set the initial base fee as per EIP-1559.
+        let base_fee_per_gas = self.initial_base_fee();
+
+        // If shanghai is activated, initialize the header with an empty withdrawals hash, and
+        // empty withdrawals list.
+        let withdrawals_root = self
+            .fork(EthereumHardfork::Shanghai)
+            .active_at_timestamp(self.genesis.timestamp)
```

Comment:
is_shanghai_active ?

Reply:
this impl is copied from `ChainSpec` so unless code was buggy before, this should work

Reply:
the issue this pr closes has nothing to do with isthmus